src/squlearn/util/pennylane/pennylane_circuit.py

```diff
@@ -0,0 +1,509 @@
+
+from qiskit.compiler import transpile
+from qiskit_aer import Aer
+
+import pennylane as qml
+import pennylane.numpy as pnp
+import pennylane.pauli as pauli
+
+from .pennylane_gates import qiskit_pennyland_gate_dict
+from ..executor import Executor
+
+
+def _get_sympy_interface():
+    """
+    Returns the sympy interface that is used in the parameter conversion.
+
+    Necessary for the correct conversion of sympy expressions in Qiskit to
+    python functions in PennyLane.
+
+    Returns:
+        Tuple of sympy printer and sympy modules
+    """
+    # SymPy printer for pennylane numpy implementation has to be set manually,
+    # otherwise math functions are used in lambdify instead of pennylane.numpy functions
+    from sympy.printing.numpy import NumPyPrinter as Printer
+
+    user_functions = {}
+    printer = Printer(
+        {
+            "fully_qualified_modules": False,
+            "inline": True,
+            "allow_unknown_functions": True,
+            "user_functions": user_functions,
+        }
+    )
+    # Use Pennylane numpy for sympy lambdify
+    modules = pnp
+
+    # The functions down below can be used to switch between different gradient engines
+    # as tensorflow, jax and torch. However, this is not supported and implemented yet.
+
+    #     # SymPy printer for pennylane numpy implementation has to be set manually,
+    #     # otherwise math functions are used in lambdify instead of pennylane.numpy functions
+    #     from sympy.printing.tensorflow import TensorflowPrinter as Printer  # type: ignore
+
+    #     user_functions = {}
+    #     printer = Printer(
+    #         {
+    #             "fully_qualified_modules": False,
+    #             "inline": True,
+    #             "allow_unknown_functions": True,
+    #             "user_functions": user_functions,
+    #         }
+    #     )  #
+    #     modules = tf
+
+    # elif self._gradient_engine == "jax":
+    #     from sympy.printing.numpy import JaxPrinter as Printer  # type: ignore
+
+    #     user_functions = {}
+    #     printer = Printer(
+    #         {
+    #             "fully_qualified_modules": False,
+    #             "inline": True,
+    #             "allow_unknown_functions": True,
+    #             "user_functions": user_functions,
+    #         }
+    #     )  #
+    #     modules = jnp
+    # elif self._gradient_engine == "torch" or self._gradient_engine == "pytorch":
+    #     from sympy.printing.pycode import PythonCodePrinter as Printer  # type: ignore
+
+    #     user_functions = {}
+    #     printer = Printer(
+    #         {
+    #             "fully_qualified_modules": False,
+    #             "inline": True,
+    #             "allow_unknown_functions": True,
+    #             "user_functions": user_functions,
+    #         }
+    #     )  #
+    #     modules = torch
+
+    # else:
+    #     # tbd for jax and tensorflow
+    #     printer = None
+    #     modules = None
```

Comment:
I would remove these comments

Reply:
I worked really hard on these :D 
In case we want to support tensorflow or so, these will be handy.